x-pack/legacy/plugins/code/model/repository.ts

```diff
@@ -56,11 +56,24 @@ export interface FileTree {
 }
 
 export function sortFileTree(a: FileTree, b: FileTree) {
-  if (a.type !== b.type) {
-    return b.type - a.type;
-  } else {
-    return a.name.localeCompare(b.name);
+  // consider Link and File are the same type, Submodule and Directory are the same type when sorting.
+  // Submodule and Directory are before Link and File
+  const types1 = [FileTreeItemType.File, FileTreeItemType.Link];
```

Comment:
for this comparator, does it make more sense to make each item in enum `FileTreeItemType` as number, so that the comparison here will be more straight-forward?

Reply:
They're already enum numbers. But to consider only compare name when two types are one of  `Link` and `File` or `Submodule` and `Directory`, it's more complex to only compare enum numbers.

Reply:
change the enum to 

```
export enum FileTreeItemType {
  File = 0,
  Link = 1,
  Submodule = 2,
  Directory = 3,
}
```
without touching this sortFileTree should fix the problem. This `sortFileTree` function's readability is not very good. We don't have tests to cover this, so let's just make simple changes.

The reason why `Link` and `Submodule` are ranked higher than `Directory` is that they currently have a higher enum value.

Reply:
But `File` and `Link` should be treated as the same type, so when doing sort, we only compare the name of it. Eventually we have to check `a.type` and `b.type` to see if they are all `Link` and `File` or the other two.

Reply:
ok, I got what you mean. thanks for the clarification.

Then I would suggest encoding the type and name into a string for sorting: e.g.

```
  const getSortName = (node: FileTree) => {
    if (node.type === FileTreeItemType.File || node.type === FileTreeItemType.Link) {
      return `1-${node.name}`;
    } else {
      return `0-${node.name}`;
    }
  };
  return getSortName(a).localeCompare(getSortName(b));
```

which seems more readable. But I will leave you to make the decision.